### PR TITLE
feat: add position API to Dialog

### PR DESCRIPTION
**Pull request by DiegoCardoso**
## Description

Adds the position API that uses the newly introduced `top` and `left` properties added to the Dialog web component.

Part of https://github.com/vaadin/web-components/issues/1060

## Type of change

- Feature

### CODE REVIEW

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java [hunk 1]
```diff
@@ -162,6 +162,18 @@ public void openDialogAddComponentAtIndex() {
         assertButtonText(1);
     }
 
+    @Test
```
Comment (sissbruecker) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java:165]: Nit: Could group this test with the other tests that use this dialog instance and that are related to resizing / dragging:

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 2]
```diff
@@ -140,6 +140,51 @@ public DialogCloseActionEvent(Dialog source, boolean fromClient) {
...
+
+    /**
+     * Sets the distance of the overlay from the left of its container. If a
+     * unitless number is provided, pixels are assumed.
+     * <p>
+     * Note that the overlay left edge may not be the same as the viewport left
+     * edge (e.g. the "Lumo" theme defines some spacing to prevent the overlay
+     * from stretching all the way to the left of the viewport).
+     */
+    public void setLeft(String left) {
```
Comment (sissbruecker) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:184]: Let's document the `left` parameter

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/dialog/tests/DialogTestPage.java [hunk 3]
```diff
@@ -192,6 +192,8 @@ private void createResizableDraggableDialog() {
         dialog.setDraggable(true);
         dialog.setWidth("200px");
         dialog.setHeight("200px");
+        dialog.setTop("50px");
+        dialog.setLeft("50px");
```
Comment (sissbruecker) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/dialog/tests/DialogTestPage.java:196]: Looks like this breaks an existing test (`draggableDialog_shouldAllowDraggingFromDivContainer`). Is this needed?

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java (hunk 1 above)
Comment (DiegoCardoso) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java:165]: Into a separate file, you mean?

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/dialog/tests/DialogTestPage.java (hunk 3 above)
Comment (DiegoCardoso) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/dialog/tests/DialogTestPage.java]: Removed.

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java (hunk 2 above)
Comment (DiegoCardoso) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:187]: Done.

File: vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java (hunk 1 above)
Comment (sissbruecker) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java:165]: No, there is a block of tests below related to the resizing / draggable dialog that is also used in this test.

Comment (DiegoCardoso) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java]: Got it. Done.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**